### ISSUE TITLE
Implement sorted() builtin

### ISSUE THREAD

**Issue by trotterdylan** [google/grumpy#16]

**Comment by MirkoDziadzka** [comment]
Just curious, is there a way in grumpy to implement a builtin function in Python?

My naïve approach of adding a Python implementation of ```sorted()```  to ```lib/__buildin__.py``` does not work (see https://github.com/MirkoDziadzka/grumpy/commit/0fe34aea09691d26a3ad6c31b4ab323638de270c)

**Comment by trotterdylan** [comment]
This limitation is briefly discussed here: https://github.com/google/grumpy/issues/50

I haven't given it a whole lot of thought but I think it's actually a bit of a pain to make this work. The basic issue is that the `__builtin__` Go package would need to import `grumpy` (the runtime package) to work properly, and the runtime package would need to import `__builtin__` so that it could populate it with the stuff implemented in Go. So there's a circular dependency.

Unfortunately, we do need to get this working because there is Python code out there that adds its own functions to `__builtin__`.

One possible approach would be to have `__builtin__.py` replace its own module `__dict__` with `__go__.grumpy.Builtins`. I don't think this would work as-is, but something along these lines:

```
# __builtin__.py
from __go__.grumpy import Builtins
import sys
sys.modules['__builtin__'].__dict__ = Builtins
```

**Comment by MirkoDziadzka** [comment]
looks like ```sorted()``` is already implemented in 384954eed28ff1eb6a0dd4c2bcc010c968a82b5a

**Comment by trotterdylan** [comment]
Whoops, sorry about that Mirko. Closing this.
